### PR TITLE
Add TimesFM Model Class with HuggingFace fine-tuning

### PR DESCRIPTION
**Pull request by arohig**
## Summary

- Migrates TimesFM from Google's native `timesfm` pip package to HuggingFace Transformers (`TimesFmModelForPrediction`, `transformers>=5.2.0`)
- Adds full fine-tuning via HF Trainer with per-window normalized loss: normalizes predictions and targets by context mean/std before MSE, preventing high-variance patients from dominating gradients
- Registers TimesFM in `ModelFactory` for the generic holdout evaluation workflow (zero-shot, fine-tune, checkpoint load, resume training)
- Gap handling pipeline matching GlucoBench: interpolate short gaps (≤45 min), segment at larger gaps, uses existing `segment_all_patients()` utility

## Changes

- **`src/models/timesfm/`**: Rewrite config and model for HF backend — `TimesFMForecaster` with HF model loading, `predict()` with NaN interpolation, batched `evaluate()`, `_train_model()` via HF Trainer with gap handling and per-patient windowing, `TimesFMForTrainer` wrapper for normalized-space loss
- **`scripts/examples/example_holdout_generic_workflow.py`**: Register `timesfm` in ModelFactory, add context-only input slicing, GPU memory management between zero-shot and fine-tuning
- **`configs/models/timesfm/fine_tune.yaml`** *(new)*: YAML config overriding workflow defaults (batch_size, num_epochs, learning_rate)
- **`pyproject.toml`**: Add `timesfm` optional dependency group
- **`scripts/setup_model_env.sh`**: Add `timesfm` to available model list

## Design Decisions

**Per-window normalized loss** — Computing loss on raw mmol/L values causes high-variance patients to dominate gradients, leading to mean reversion. The `TimesFMForTrainer` wrapper normalizes both predictions and targets by context mean/std before MSE.

**Gap handling pipeline** — Uses `segment_all_patients()` to interpolate short CGM gaps (≤45 min) and segment at larger gaps before windowing. Matches the GlucoBench benchmark pipeline.

**Per-patient windowing** — Sliding windows are created independently within each patient/segment, ensuring no window crosses boundaries. Train/val split is by patient ID (80/20).

## Notes

- Requires `transformers>=5.2.0` (TimesFM was added to HF Transformers in v5.2.0)
- HF TimesFM only accepts univariate `past_values` — covariates (IOB, COB) require Google's native `timesfm` package with XReg
- Requires GPU with ≥24GB VRAM for fine-tuning

## Testing

```bash
# Setup environment
source scripts/setup_model_env.sh timesfm

# Fine-tune on brown_2019 (SLURM)
srun --gres=gpu:1 --mem=32G --time=2:00:00 \
  python scripts/examples/example_holdout_generic_workflow.py \
    --model-type timesfm \
    --model-config configs/models/timesfm/fine_tune.yaml \
    --datasets brown_2019 \
    --config-dir configs/data/holdout_5pct \

# Zero-shot only
srun --gres=gpu:1 --mem=32G --time=1:00:00 \
  python scripts/examples/example_holdout_generic_workflow.py \
    --model-type timesfm \
    --datasets brown_2019 \
    --config-dir configs/data/holdout_5pct \
    --skip-training
```

- Verify steps 4 (zero-shot) → 5 (fine-tune) → 6 (load checkpoint) → 7 (resume training) → 8 (final evaluation) succeed
- Verify `--skip-training` reuses zero-shot model for step 8

## Validation Results

Trained on brown_2019 dataset (10 epoch fine-tuning):

| Holdout | RMSE (mean) | RMSE (median) | Episodes |
|---------|-------------|---------------|----------|
| Patient  | 2.062 mM | 1.617 mM | 500 |
| Temporal | 2.076 mM | 1.756 mM | 500 |

### CODE REVIEW

**Review by RobotPsychologist — APPROVED** [review]
LGTM
